vector/src/main/java/im/vector/app/features/spaces/explore/SpaceDirectoryViewModel.kt

```diff
@@ -93,29 +96,44 @@ class SpaceDirectoryViewModel @AssistedInject constructor(
                 .disposeOnClear()
     }
 
-    private fun refreshFromApi() {
+    private fun refreshFromApi(rootId: String?) = withState { state ->
+        val spaceId = rootId ?: initialState.spaceId
         setState {
             copy(
-                    spaceSummaryApiResult = Loading()
+                    apiResults = state.apiResults.toMutableMap().apply {
+                        this[spaceId] = Loading()
+                    }.toMap()
             )
         }
         viewModelScope.launch(Dispatchers.IO) {
+            val cachedResults = state.apiResults.toMutableMap()
             try {
-                val query = session.spaceService().querySpaceChildren(initialState.spaceId)
-                val knownSummaries = query.second.mapNotNull {
+                val query = session.spaceService().querySpaceChildren(
+                        spaceId,
+                        limit = 10
+                )
+                val knownSummaries = query.children.mapNotNull {
                     session.getRoomSummary(it.childRoomId)
                             ?.takeIf { it.membership == Membership.JOIN } // only take if joined because it will be up to date (synced)
-                }
+                }.distinctBy { it.roomId }
```

Comment:
Why adding this call to `distinctBy()`? The server can send duplicates? If it is the case, it's probably better to handle that SDK side.